src/access/examples/ExampleAccessControlDelegate.cpp

```diff
@@ -0,0 +1,1101 @@
+
+    // ((mCluster & kClusterIdMask) < kClusterIdMinMs) --> invalid
+    static constexpr ClusterId kClusterIdMinMs = 0x0000FC00;
+
+    // ((mCluster & kClusterIdMask) < kClusterIdMaxMs) --> invalid
+    static constexpr ClusterId kClusterIdMaxMs = 0x0000FFFE;
+
+    // (mCluster & kClusterVendorMask) --> cluster vendor portion
+    static constexpr ClusterId kClusterVendorMask = 0xFFFF0000;
+
+    // ((mCluster & kClusterVendorMask) < kClusterVendorMin) --> invalid
+    static constexpr ClusterId kClusterVendorMin = 0x00000000;
+
+    // ((mCluster & kClusterVendorMask) > kClusterVendorMax) --> invalid
+    static constexpr ClusterId kClusterVendorMax = 0xFFFE0000;
+
+    // (mDeviceType == kDeviceTypeEmpty) --> mDeviceType contains neither endpoint nor device type
+    static constexpr DeviceTypeId kDeviceTypeEmpty = 0xFFFFFFFF;
+
+    // (mDeviceType & kDeviceTypeIdMask) --> device type id portion
+    static constexpr DeviceTypeId kDeviceTypeIdMask = 0x0000FFFF;
+
+    // ((mDeviceType & kDeviceTypeIdMask) < kDeviceTypeIdMin) --> invalid
+    static constexpr DeviceTypeId kDeviceTypeIdMin = 0x00000000;
+
+    // ((mDeviceType & kDeviceTypeIdMask) < kDeviceTypeIdMax) --> invalid
+    static constexpr DeviceTypeId kDeviceTypeIdMax = 0x0000BFFF;
+
+    // (mDeviceType & kDeviceTypeVendorMask) --> device type vendor portion
+    static constexpr DeviceTypeId kDeviceTypeVendorMask = 0xFFFF0000;
+
+    // ((mDeviceType & kDeviceTypeVendorMask) < kDeviceTypeVendorMin) --> invalid
+    static constexpr DeviceTypeId kDeviceTypeVendorMin = 0x00000000;
+
+    // ((mDeviceType & kDeviceTypeVendorMask) > kDeviceTypeVendorMax) --> invalid
+    static constexpr DeviceTypeId kDeviceTypeVendorMax = 0xFFFE0000;
+
+    // ((mDeviceType & kDeviceTypeIdMask) == kEndpointMagic) --> mDeviceType contains endpoint
+    static constexpr DeviceTypeId kEndpointMagic = 0x0000EEEE;
+
+    // (mDeviceType >> kEndpointShift) --> extract endpoint from mDeviceType
+    static constexpr int kEndpointShift = 16;
+
+    static_assert(kClusterEmpty == kDeviceTypeEmpty);
+    static_assert(int(kClusterEmpty) == (kEmptyByte | kEmptyByte << 8 | kEmptyByte << 16 | kEmptyByte << 24));
+
+private:
+    ClusterId mCluster;
+    DeviceTypeId mDeviceType;
+};
+
+class EntryStorage
+{
+public:
+    // ACL support
+    static EntryStorage acl[kNumberOfFabrics * kEntriesPerFabric];
+
+    static EntryStorage * FindUnusedInAcl(size_t * index, const FabricIndex * fabricIndex)
+    {
+        for (auto & storage : acl)
+        {
+            if (!storage.InUse())
+            {
+                if (index != nullptr)
+                {
+                    if (fabricIndex != nullptr)
+                    {
+                        *index = 0;
+                        for (auto & storage2 : acl)
+                        {
+                            if (&storage2 == &storage)
+                            {
+                                break;
+                            }
```

Comment:
It was looking up the fabric filtered index, if requested as an out parameter. But I've since refactored all the index conversion (from absolute index in ACL, to index relative to fabric) into one function, so it's not here any more. This conversation should be resolvable now.